typing/mode.ml

```diff
@@ -2295,37 +2335,29 @@ module Value_with (Areality : Areality) = struct
 
   type lr = (allowed * allowed) t
 
-  type ('m, 'a, 'd) axis =
-    | Monadic :
-        (Monadic.Const.t, 'a) Axis.t
-        -> (('a, 'd) mode_monadic, 'a, 'd) axis
-    | Comonadic :
-        (Comonadic.Const.t, 'a) Axis.t
-        -> (('a, 'd) mode_comonadic, 'a, 'd) axis
-
-  type 'd axis_packed = P : ('m, 'a, 'd) axis -> 'd axis_packed
-
-  let print_axis (type m a d) ppf (axis : (m, a, d) axis) =
-    match axis with
-    | Monadic ax -> Axis.print ppf ax
-    | Comonadic ax -> Axis.print ppf ax
-
-  let lattice_of_axis (type m a d) (axis : (m, a, d) axis) :
-      (module Lattice with type t = a) =
-    match axis with
-    | Comonadic ax -> Comonadic.Const.lattice_of_axis ax
-    | Monadic ax -> Monadic.Const.lattice_of_axis ax
-
-  let all_axes =
-    [ P (Comonadic Areality);
-      P (Monadic Uniqueness);
-      P (Comonadic Linearity);
-      P (Monadic Contention);
-      P (Comonadic Portability);
-      P (Comonadic Statefulness);
-      P (Monadic Visibility) ]
-
-  let proj_obj : type m a d. (m, a, d) axis -> a C.obj = function
+  module Axis = struct
+    type ('a, 'd0, 'd1) t =
+      | Monadic : (Monadic.Const.t, 'a) Axis.t -> ('a, 'd, 'd neg) t
+      | Comonadic : (Comonadic.Const.t, 'a) Axis.t -> ('a, 'd, 'd pos) t
```

Comment:
TBH the new GADT doesn't seem much simpler than before, but at least it no longer mentions `mode_monadic` and `mode_comonadic`, and instead uses allowance to encode flipping.